### PR TITLE
Improved easyconfigs for GROMACS 2024.1 with GPUs

### PR DESCRIPTION
**Pull request by mszpindler**

### CODE REVIEW

File: easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeAMD-23.09-rocm-5.4.6.eb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+whatis = [
+    "Description: Implementation of SYCL and C++ standard parallelism for CPUs and GPUs from all vendors: The independent, community-driven compiler for C++-based heterogeneous programming models."
+]
+
+description = """
+AdaptiveCpp is the independent, community-driven modern platform for C++-based heterogeneous programming models targeting CPUs and GPUs from all major vendors. 
+"""
+
+toolchain = {'name': 'cpeAMD', 'version': '23.09'}
+toolchainopts = {'verbose': False, 'openmp': True}
```
Comment (olouant) [inline easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeAMD-23.09-rocm-5.4.6.eb:18]:
```suggestion
toolchainopts = {'verbose': False, 'openmp': False}
```
During my tests, I noticed that building ACPP manually did not present the missing symbol problem. The main difference in my manual build was the absence of the `-fopenmp` flag in the `CXXFLAGS` environment variable.

I suggest disabling OpenMP here as it has not influence on the build of the OpenMP backend.

File: easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeAMD-23.09-rocm-5.4.6.eb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+preconfigopts  = 'module unload amd && '
+preconfigopts += 'module load rocm/5.4.6 && '
+preconfigopts += 'export PATH=${ROCM_PATH}/llvm/bin:$PATH && '
+preconfigopts += 'export C_INCLUDE_PATH=$ROCM_PATH/llvm/include && '
+preconfigopts += 'export CPLUS_INCLUDE_PATH=$ROCM_PATH/llvm/include && '
+preconfigopts += 'export CMAKE_LIBRARY_PATH=${ROCM_PATH}/llvm/lib:${ROCM_PATH}/lib:$CMAKE_LIBRARY_PATH && '
+preconfigopts += 'export CMAKE_PREFIX_PATH=${ROCM_PATH}/hip:${ROCM_PATH}:$CMAKE_PREFIX_PATH && '
+preconfigopts += 'export LIBRARY_PATH=${ROCM_PATH}/llvm/lib:${EBROOTBOOST}/lib:${EBROOTBUILDTOOLS}/lib:${ROCM_PATH}/lib && '
+preconfigopts += 'export LD_LIBRARY_PATH=${ROCM_PATH}/hsa/lib:${ROCM_PATH}/lib64:${ROCM_PATH}/lib:${ROCM_PATH}/llvm/lib:$LD_LIBRARY_PATH && '
+preconfigopts += 'export CRAY_LD_LIBRARY_PATH=${ROCM_PATH}/hsa/lib:${ROCM_PATH}/lib64:${ROCM_PATH}/lib:${ROCM_PATH}/llvm/lib:$CRAY_LD_LIBRARY_PATH && '
```
Comment (olouant) [inline easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeAMD-23.09-rocm-5.4.6.eb:49]:
```suggestion
```

File: easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeAMD-23.09-rocm-5.4.6.eb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+prebuildopts  = 'module unload amd && '
+prebuildopts += 'module load rocm/5.4.6 && '
+prebuildopts += 'export PATH=${ROCM_PATH}/llvm/bin:$PATH && '
+prebuildopts += 'export C_INCLUDE_PATH=$ROCM_PATH/llvm/include && '
+prebuildopts += 'export CPLUS_INCLUDE_PATH=$ROCM_PATH/llvm/include && '
+prebuildopts += 'export CMAKE_LIBRARY_PATH=${ROCM_PATH}/llvm/lib:${ROCM_PATH}/lib:$CMAKE_LIBRARY_PATH && '
+prebuildopts += 'export CMAKE_PREFIX_PATH=${ROCM_PATH}/hip:${ROCM_PATH}:$CMAKE_PREFIX_PATH && '
+prebuildopts += 'export LIBRARY_PATH=${ROCM_PATH}/llvm/lib:${EBROOTBOOST}/lib:${EBROOTBUILDTOOLS}/lib:${ROCM_PATH}/lib && '
+prebuildopts += 'export LD_LIBRARY_PATH=${ROCM_PATH}/hsa/lib:${ROCM_PATH}/lib64:${ROCM_PATH}/lib:${ROCM_PATH}/llvm/lib:$LD_LIBRARY_PATH && '
+prebuildopts += 'export CRAY_LD_LIBRARY_PATH=${ROCM_PATH}/hsa/lib:${ROCM_PATH}/lib64:${ROCM_PATH}/lib:${ROCM_PATH}/llvm/lib:$CRAY_LD_LIBRARY_PATH && '
```
Comment (olouant) [inline easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeAMD-23.09-rocm-5.4.6.eb:82]:
```suggestion
```

**Comment by mszpindler** [comment]
```
PERFORMANCE REPORT
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
[lumi_gromacs_stmv %benchmark_info=stmv_v2 %nb_impl=gpu %update_mode=gpu %bonded_impl=gpu %fft_variant=heffte %num_nodes=1 /c80fd3eb @lumi:gpu:cpeAMD]
  num_tasks: 8
  use_multithreading: False
  num_gpus_per_node: 8
  num_tasks_per_node: 8
  performance:
    - perf: 100.477 ns/day (r: 70.0 ns/day l: -5.0% u: +inf%)
  ```
  Similar numbers for VkFFT variant.

**Pull request merged**